src/PowerShellEditorServices/Services/DebugAdapter/Handlers/LaunchAndAttachHandler.cs

```diff
@@ -222,10 +247,19 @@ await _executionService.ExecutePSCommandAsync(
             if (_debugStateService.ScriptToLaunch != null
                 && _runspaceContext.CurrentRunspace.IsOnRemoteMachine)
             {
-                _debugStateService.ScriptToLaunch =
-                    _remoteFileManagerService.GetMappedPath(
-                        _debugStateService.ScriptToLaunch,
-                        _runspaceContext.CurrentRunspace);
+                if (_debugService.TryGetMappedRemotePath(_debugStateService.ScriptToLaunch, out string remoteMappedPath))
+                {
+                    _debugStateService.ScriptToLaunch = remoteMappedPath;
+                }
+                else
+                {
+                    // If the script is not mapped, we will map it to the remote path
+                    // using the RemoteFileManagerService.
+                    _debugStateService.ScriptToLaunch =
```

Comment:
Can you simplify this so the assignment to ScriptToLaunch happens only once, with the if/else covering the two logical paths to get the (err) path?